src/main/java/seedu/address/logic/commands/DeleteAssignmentCommand.java

```diff
@@ -45,14 +48,25 @@ public DeleteAssignmentCommand(Name name, Index targetIndex) {
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
 
-        List<Assignment> lastShownList = model.getFilteredAssignmentList(name);
+        // Get Person that match the name
+        List<Person> filteredPersonList =
+                model.getFilteredPersonList()
+                        .stream()
+                        .filter(person -> person.isSameName(name))
+                        .collect(Collectors.toList());
+
+        if (filteredPersonList.size() == 0) {
+            throw new CommandException(MESSAGE_INVALID_PERSON_DISPLAYED_NAME);
+        }
+        Person selectedPerson = filteredPersonList.get(0);
+        List<Assignment> lastShownList = model.getFilteredAssignmentList(selectedPerson);
 
         if (lastShownList.size() == 0) {
-            throw new CommandException(Messages.MESSAGE_ASSIGNMENTS_EMPTY);
+            throw new CommandException(Messages.MESSAGE_INVALID_ASSIGNMENT_DISPLAYED_INDEX);
         }
```

Comment:
If the lastShownList.size() < the number represented by the index, then the assignment displayed index is invalid, i mean we should ensure the index is not negative and then compare the size of lastShownList and index to decide whether the index is valid or not.